packages/core/src/external/carequality/ihe-gateway-v2/xca/process-dq-response.ts

```diff
@@ -0,0 +1,171 @@
+import { handleRegistryErrorResponse, handleHTTPErrorResponse, handleEmptyResponse } from "./error";
+import { DQSamlClientResponse } from "../saml-client";
+import { stripUrnPrefix } from "../utils";
+
+type Identifier = {
+  _identificationScheme: string;
+  _value: string;
+};
+
+type Classification = {
+  _classificationScheme: string;
+  Name: {
+    LocalizedString: {
+      _charset: string;
+      _value: string;
+    };
+  };
+};
+
+type Slot = {
+  _name: string;
+  ValueList: {
+    Value: string | string[];
+  };
+};
+
+function parseDocumentReference(
+  //eslint-disable-next-line @typescript-eslint/no-explicit-any
+  extrinsicObject: any
+): DocumentReference {
+  const slots = extrinsicObject?.Slot;
+  const externalIdentifiers = extrinsicObject?.ExternalIdentifier;
+  const classifications = extrinsicObject?.Classification;
+
+  const findSlotValue = (name: string): string => {
+    const slot = slots.find((slot: Slot) => slot._name === name);
+    return slot
+      ? Array.isArray(slot.ValueList.Value)
+        ? slot.ValueList.Value.join(", ")
+        : slot.ValueList.Value
+      : undefined;
+  };
+
+  const findExternalIdentifierValue = (scheme: string) => {
+    const identifier = externalIdentifiers?.find(
+      (identifier: Identifier) => identifier._identificationScheme === scheme
+    );
+    return identifier ? identifier._value : undefined;
+  };
+
+  const findClassificationSlotValue = (
+    classificationScheme: string,
+    slotName: string
+  ): string | undefined => {
+    const classification = classifications.find(
+      (c: Classification) => c._classificationScheme === classificationScheme
+    );
+    if (!classification) return undefined;
+    const slot = classification.Slot?.find((s: Slot) => s._name === slotName);
+    return slot
+      ? Array.isArray(slot.ValueList.Value)
+        ? slot.ValueList.Value.join(", ")
+        : slot.ValueList.Value
+      : undefined;
+  };
+
+  const findClassificationName = (scheme: string) => {
+    const classification = classifications?.find(
+      (classification: Classification) => classification?._classificationScheme === scheme
+    );
+    if (!classification) return undefined;
+    const title = classification?.Name?.LocalizedString?._value;
+    return title;
+  };
+
+  const documentReference: DocumentReference = {
+    homeCommunityId: stripUrnPrefix(extrinsicObject._home),
+    repositoryUniqueId: findSlotValue("repositoryUniqueId"),
+    docUniqueId: findExternalIdentifierValue("urn:uuid:2e82c1f6-a085-4c72-9da3-8640a32e42ab"),
+    contentType: extrinsicObject?._mimeType,
+    language: findSlotValue("languageCode"),
+    size: parseInt(findSlotValue("size")),
+    title: findClassificationName("urn:uuid:41a5887f-8865-4c09-adf7-e362475b143a"),
+    creation: findSlotValue("creationTime"),
+    authorInstitution: findClassificationSlotValue(
+      "urn:uuid:93606bcf-9494-43ec-9b4e-a7748d1a838d",
+      "authorInstitution"
```

Comment:
wtf do all these UUIDs mean?

suggest putting them in an appropriately named const for readability, and so we can re-use later if needed

Reply:
These are the UUIDs that are always used for the specific values in the DR. So for example for authorInstitution: 
https://gazelle.ihe.net/XDStarClient/rimihe-templates/rimihe-Author_XDSDocumentEntry.html

Good call on the constants though will refactor